adb_shell/adb_device.py

```diff
@@ -398,9 +398,42 @@ def _service(self, service, command, timeout_s=None, total_timeout_s=constants.D
 
         """
         adb_info = _AdbTransactionInfo(None, None, timeout_s, total_timeout_s)
+        stream = self._streaming_command(service, command, adb_info)
         if decode:
-            return b''.join(self._streaming_command(service, command, adb_info)).decode('utf8')
-        return b''.join(self._streaming_command(service, command, adb_info))
+            return b''.join(stream).decode('utf8')
+        return b''.join(stream)
+
+    def _streaming_service(self, service, command, timeout_s=None, total_timeout_s=constants.DEFAULT_TOTAL_TIMEOUT_S, decode=True):
+        """Send an ADB command to the device.
```

Comment:
Please add something to this method's description to distinguish it from `_service`. How about:

Send an ADB command to the device, yielding each line of output.